KanatStore/KanatStore.BL/Dto/ProductDto.cs

```diff
@@ -1,34 +1,39 @@
 ﻿using System.ComponentModel.DataAnnotations;
 using System.ComponentModel;
+using Microsoft.AspNetCore.Http;
+using System.ComponentModel.DataAnnotations.Schema;
 
 namespace KanatStore.BLL.Dto
 {
     public class ProductDto
     {
         [Key]
         public int Id { get; set; }
-        [Required]
+        [Required(ErrorMessage ="Tên sản phẩm không được bỏ trống")]
         [DisplayName("Tên sản phẩm")]
         public string Name { get; set; }
         [DisplayName("Chiều dài")]
+        [Required(ErrorMessage = "Chiều dài không được bỏ trống")]
         public float Length { get; set; }
+        [Required(ErrorMessage = "Chiều rộng không được bỏ trống")]
         [DisplayName("Chiều rộng")]
         public float Width { get; set; }
         [DisplayName("Chiều dày")]
-        public float Thickness { get; set; }
+        public float? Thickness { get; set; }
         [DisplayName("Xuất xứ")]
-        public string Origin { get; set; }
+        public string? Origin { get; set; }
```

Comment:
Bản thân string có thể nhận giá trị là null => string? là không cần thiết (không có ý nghĩa)

Reply:
Nhưng khi em không để string?, lúc e không nhập dữ liệu cho ô đó thì nó báo lỗi "The Xuất xứ field is required."